api/rbac/rbac.go

```diff
@@ -121,27 +122,43 @@ func GetUser(c echo.Context) (string, error) {
 	return subject, nil
 }
 
+// buildPathResourceMap builds a map of paths to resources and a list of resources.
+// Returns: (resourceMap, skipPaths, error) .
+func buildPathResourceMap(basePath string) (map[string]string, []string, error) {
+	swg, err := everestclient.GetSwagger()
+	if err != nil {
+		return nil, nil, errors.Join(err, errors.New("failed to get swagger"))
+	}
+
+	// parseEndpoint replaces the curly braces in the endpoint with colons.
+	// example: '/{namespace}/clusters' -> '/:namespace/clusters'
+	parseEndpoint := func(ep string) string {
+		parsed := strings.ReplaceAll(ep, "{", ":")
+		parsed = strings.ReplaceAll(parsed, "}", "")
+		return basePath + parsed
+	}
+
+	resourceMap := make(map[string]string)
+	skipPaths := []string{}
+	for path, pathItem := range swg.Paths.Map() {
+		parsedPath := parseEndpoint(path)
+		if val, ok := pathItem.Extensions[common.EverestAPIExtnResourceName]; ok {
+			if resourceName, ok := val.(string); ok {
+				resourceMap[parsedPath] = resourceName
+			}
+			continue
+		}
+		skipPaths = append(skipPaths, parsedPath)
```

Comment:
So starting for now we shouldn't forget to add the `x-everest-resource-name` extension to the new endpoints otherwise they won't be authenticated. It's not that obvious so I was thinking if we can make it more prominent somehow

Reply:
This extension controls authorisation only, authentication is controlled using the `security: []` property

But yeah, its still possible to forget to add it if we add new APIs. I thought of adding some manual CI checks, but the problem there is same; how would it know which path needs RBAC and which doesn't? 🤔

Reply:
But with the previous approach, we had to remember to modify code, so I still this this is a decent improvement from that 😅 
I'm happy to discuss ideas to further improve it..

Reply:
I don't think there's anything we can do about it, we need to manually control this either way.

Reply:
Makes sense, let's just keep it in mind when adding and reviewing new endpoints 👍